### PR TITLE
GDiceLoss(V2) fixes

### PR DESCRIPTION
**Pull request by clementpoiret**
Hi!

Here is a quick fix for GDiceLoss and GDiceLossV2.
The original code is expecting a function to apply_nonlin, and if apply_nonlin is None (default), softmax_output is never assigned, even if it is called later in the loss function.

I hope the changes are ok for you, as it should work correctly without nonlin now :) !
Have a nice day,
Clément

### CODE REVIEW

**Comment by clementpoiret** [comment]
The problem is also here for SSLoss, I fixed it in my second commit :+1:

**Comment by JunMa11** [comment]
Thank you so much:)
